lib/rules/jsx-prefer-fragment-shorthand.js

```diff
@@ -0,0 +1,119 @@
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+function replaceNode(source, node, text) {
+  return `${source.slice(0, node.range[0])}${text}${source.slice(node.range[1])}`;
+}
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Enforce standard or shorthand syntax for React fragments',
+      category: 'Stylistic Issues',
+      recommended: false,
+      url: docsUrl('jsx-prefer-fragment-shorthand')
+    },
+    fixable: 'code',
+
+    schema: [{
+      enum: ['always', 'never']
+    }]
+  },
+
+  create: function(context) {
+    const configuration = context.options[0] || 'always';
+    const sourceCode = context.getSourceCode();
+    const reactPragma = pragmaUtil.getFromContext(context);
+    const fragmentPragma = pragmaUtil.getFragmentFromContext(context);
+    const openFragShort = '<>';
+    const closeFragShort = '</>';
+    const openFragLong = `<${reactPragma}.${fragmentPragma}>`;
+    const closeFragLong = `</${reactPragma}.${fragmentPragma}>`;
+
+    function reportOnReactVersion(node) {
+      if (!versionUtil.testReactVersion(context, '16.2.0')) {
+        context.report({
+          node,
+          message: 'Fragments are only supported starting from React v16.2'
+        });
+        return true;
+      }
+
+      return false;
+    }
+
+    function getFixerToLong(jsxFragment) {
+      return function(fixer) {
+        let source = sourceCode.getText();
+        source = replaceNode(source, jsxFragment.closingFragment, closeFragLong);
+        source = replaceNode(source, jsxFragment.openingFragment, openFragLong);
+        const lengthDiff = openFragLong.length - sourceCode.getText(jsxFragment.openingFragment).length
+          + closeFragLong.length - sourceCode.getText(jsxFragment.closingFragment).length;
+        const range = jsxFragment.range;
+        return fixer.replaceTextRange(range, source.slice(range[0], range[1] + lengthDiff));
+      };
+    }
+
+    function getFixerToShort(jsxElement) {
+      return function(fixer) {
+        let source = sourceCode.getText();
+        source = replaceNode(source, jsxElement.closingElement, closeFragShort);
+        source = replaceNode(source, jsxElement.openingElement, openFragShort);
+        const lengthDiff = sourceCode.getText(jsxElement.openingElement).length - openFragShort.length
+          + sourceCode.getText(jsxElement.closingElement).length - closeFragShort.length;
+        const range = jsxElement.range;
+        return fixer.replaceTextRange(range, source.slice(range[0], range[1] - lengthDiff));
+      };
+    }
+
+    // --------------------------------------------------------------------------
+    // Public
+    // --------------------------------------------------------------------------
+
+    return {
+      JSXElement(node) {
+        const openingEl = node.openingElement;
+        if (elementType(openingEl) === `${reactPragma}.${fragmentPragma}`) {
```

Comment:
What if someone already has `<Fragment`? I’d expect to warn on that, if it was destructured from React.

Reply:
Oh. That’s what I was trying to figure out in #1661, but I took https://github.com/yannickcr/eslint-plugin-react/issues/1661#issuecomment-423386464 to mean we don’t have to handle that for now?

Reply:
I don’t think we have to handle it in the autofixer - but i think we do need to handle it in terms of warning.

Reply:
Also
```js
const { Fragment } = require('react');
```